rpkg_src/extract_gfxf_from.cpp

```diff
@@ -27,583 +29,598 @@ void rpkg_function::extract_gfxf_from(std::string& input_path, std::string& filt
-                                    position += 0x4;
+                                std::ofstream gfx_output_file = std::ofstream(
+                                    gfxf_output_dir + "\\" + hash_file_name + ".GFX", std::ofstream::binary);
 
-                                    //std::cout << bytes4 << std::endl;
-                                    //std::cout << gfx_file_length << std::endl;
+                                if (!gfx_output_file.good())
+                                {
+                                    LOG_AND_EXIT(
+                                        "Error: DDS file " + gfxf_output_dir + "\\" + hash_file_name + ".GFX" +
+                                        " could not be created.");
+                                }
 
-                                    if (bytes4 != gfx_file_length)
-                                    {
-                                        LOG(hash_file_name + " does not have a valid BIN1 IOI header.");
-                                    }
+                                gfx_output_file.write(gfx_data.data(), gfx_file_length);
 
-                                    std::vector<char> gfx_data;
+                                gfx_output_file.close();
 
-                                    for (uint64_t k = 0; k < gfx_file_length; k++)
-                                    {
-                                        gfx_data.push_back(gfxf_data->data()[position]);
+                                if (gfxf_contains_dds)
+                                {
+                                    uint32_t dds_names_count = 0;
+                                    std::vector<std::string> dds_names;
+                                    std::vector<uint32_t> dds_names_lengths;
 
+                                    while (((position & 0xF) != 0x4) && ((position & 0xF) != 0xC))
+                                    {
                                         position++;
                                     }
 
-                                    std::string gfxf_output_dir = file::output_path_append("GFXF\\" + rpkgs.at(i).rpkg_file_name + "\\" + hash_file_name, output_path);
-
-                                    file::create_directories(gfxf_output_dir);
+                                    std::memcpy(&dds_names_count, &gfxf_data->data()[position], sizeof(bytes4));
 
-                                    std::ofstream gfx_output_file = std::ofstream(gfxf_output_dir + "\\" + hash_file_name + ".GFX", std::ofstream::binary);
-
-                                    if (!gfx_output_file.good())
+                                    for (uint64_t k = 0; k < sizeof(bytes4); k++)
                                     {
-                                        LOG_AND_EXIT("Error: DDS file " + gfxf_output_dir + "\\" + hash_file_name + ".GFX" + " could not be created.");
+                                        gfxf_meta_data.push_back(gfxf_data->data()[position + k]);
```

Comment:
Did you mean to replace LOG_AND_EXIT with that?

Reply:
Not actually changed, just looks like it due to whitespace.